File: agent-dao/src/main/java/com/agent/platform/dao/entity/app/AppTemplate.java
package com.agent.platform.dao.entity.app;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 应用模板（tenant_id=0 表示平台内置模板）
 */
@Data
@TableName("app_template")
public class AppTemplate {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long tenantId;

    /** 模板名称 */
    private String name;

    /** 分类：customer-service/translate/content/data-analysis/marketing/coding/custom */
    private String category;

    /** 应用类型：chatflow/workflow/agent */
    private String appType;

    /** 图标（emoji） */
    private String icon;

    /** 模板简介 */
    private String description;

    /** 适用场景 */
    private String useCase;

    /** 创建应用后的默认开场白 */
    private String welcomeMessage;

    /** 是否平台内置：0否 1是 */
    private Integer builtin;

    /** 被使用次数 */
    private Integer usageCount;

    /** 状态：0停用 1启用 */
    private Integer status;

    /** 创建人 */
    private Long createBy;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: frontend/src/api/app-template.ts
import request from './request'
import type { AppAgent, PageResult } from './types'

/** 应用模板 */
export interface AppTemplate {
  id: number
  tenantId: number
  name: string
  /** customer-service/translate/content/data-analysis/marketing/coding/custom */
  category?: string
  /** chatflow/workflow/agent */
  appType: string
  /** 图标（emoji） */
  icon?: string
  description?: string
  useCase?: string
  welcomeMessage?: string
  /** 是否平台内置：0否 1是 */
  builtin: number
  usageCount: number
  /** 状态：0停用 1启用 */
  status: number
  createTime?: string
  updateTime?: string
}

export interface TemplatePayload {
  name: string
  category?: string
  appType?: string
  icon?: string
  description?: string
  useCase?: string
  welcomeMessage?: string
  status?: number
}

/** 应用模板 API（/api/app/templates） */
export const appTemplateApi = {
  page(params: { page?: number; size?: number; keyword?: string; category?: string; appType?: string }) {
    return request.get<never, PageResult<AppTemplate>>('/app/templates', { params })
  },
  create(data: TemplatePayload) {
    return request.post<never, AppTemplate>('/app/templates', data)
  },
  update(id: number, data: TemplatePayload) {
    return request.put<never, void>(`/app/templates/${id}`, data)
  },
  remove(id: number) {
    return request.delete<never, void>(`/app/templates/${id}`)
  },
  /** 从模板一键创建应用草稿 */
  instantiate(id: number, name?: string) {
    return request.post<never, AppAgent>(`/app/templates/${id}/instantiate`, name ? { name } : {})
  }
}
